lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2526,7 +2526,7 @@ static bool isVisibleFromModule(const ClangModuleUnit *ModuleFilter,
   // Handle redeclarable Clang decls by checking each redeclaration.
   bool IsTagDecl = isa<clang::TagDecl>(D);
   if (!(IsTagDecl || isa<clang::FunctionDecl>(D) || isa<clang::VarDecl>(D) ||
-        isa<clang::TypedefNameDecl>(D))) {
+        isa<clang::TypedefNameDecl>(D) || isa<clang::NamespaceDecl>(D))) {
```

Comment:
How'd you discover these weren't being marked as visible? Did you add a test/can you?

Reply:
I initially changed this because module interfaces for `std` submodules didn't include `namespace std` redeclarations, but now if I remove this, it still works correctly... I'll drop this part of the change, sorry for that.